### ISSUE TITLE
Support string => []string using some sort of delimiter

### ISSUE THREAD

**Issue by chriskolenko** [moogar0880/venom#16]
It would be nice if I could using an environment variable like so:
"APIKEYS" = "key1,key2,key3"
And unmarshal to a list of keys.

**Comment by moogar0880** [comment]
Yeah, I can see this being useful since environment variables make it sort of difficult to express things like arrays in any other way. 

I guess the part of this that makes things particularly difficult is that, by the time a value gets into the reflection logic that handles coercing it into it's destination type, the decoder no longer has a notion of what level the config came from. We could maybe add something to the `coerce*Slice` functions that allow them to handle converting from a `string` instance, and then rely on a CSV reader to convert them to slices? 

I'll play around with it a bit and see what I can come up with.
